pkg/policy/proportion/proportion.go

```diff
@@ -103,6 +110,29 @@ func (ps *proportionScheduler) Allocate(
 			}
 		}
 	}
+
+	// assign the left resources to queues one by one
+	leftCPU := totalCPU - totalAllocatedCPU
+	leftMEM := totalMEM - totalAllocatedMEM
+	for _, queue := range allocatedResult {
+		resList := queue.Queue().Status.Deserved.DeepCopy()
+		if leftCPU > 0 {
+			leftCPU -= 1
+			result := resList.Resources["cpu"].DeepCopy()
+			result.Add(*resource.NewQuantity(1, resource.DecimalSI))
+			queue.Queue().Status.Deserved.Resources["cpu"] = result
+		}
+		if leftMEM > 0 {
+			leftMEM -= 1
+			result := resList.Resources["memory"].DeepCopy()
+			result.Add(*resource.NewQuantity(1, resource.BinarySI))
+			queue.Queue().Status.Deserved.Resources["memory"] = result
+		}
+	}
+	if leftCPU > 0 || leftMEM > 0 {
+		glog.Errorf("fail to assign the left resources, left cpu %d, left memory %d", leftCPU, leftMEM)
```

Comment:
we should make sure all resources are allocated instead of an error message.

Reply:
The `leftCPU` and `leftMEM` is less than the size of `allocatedResult`, so all of them can be allocated in above loop. I will remove the error message.